meson.build

```diff
@@ -756,20 +756,33 @@ if has_semihost
 
   picolibc_link_path = '@0@/@1@'.format(meson.current_source_dir(), 'picolibc.ld')
 
-  test_c_args += ['--specs', '@0@/@1@'.format(meson.current_build_dir(), picolibc_specs)]
+  # Clang does not support spec files
+  if meson.get_compiler('c').get_id() != 'clang'
+    test_c_args += ['--specs', '@0@/@1@'.format(meson.current_build_dir(), picolibc_specs)]
+  endif
 
   # Undefine _exit as that's in libsemihost and needs to replace the weak _exit
 
-  test_link_args += test_c_args + ['-nostartfiles', '-nostdlib', '-T', test_link_path, '-T', picolibc_link_path, '-Wl,--undefined=_exit', '-lgcc']
+  test_link_args += test_c_args
+
+  # Clang does not support -nostartfiles
+  if meson.get_compiler('c').get_id() != 'clang'
+    test_link_args += '-nostartfiles'
```

Comment:
`-nostdlib` implies `-nostartfiles`, so this can be removed for both clang and GCC.

Source: https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html